applications/ParticleMechanicsApplication/python_scripts/deprecated_python_scripts/deprecated_particle_mpm_solver.py

```diff
@@ -0,0 +1,370 @@
+                    self.new_element = KratosParticle.CreateUpdatedLagragian2D4N()
+            else:
+                if (self.pressure_dofs):
+                    raise Exception("Element for mixed U-P formulation in 3D for Hexahedral Element is not yet implemented.")
+                else:
+                    self.new_element = KratosParticle.CreateUpdatedLagragian3D8N()
+
+        # Initialize solver
+        if(self.domain_size==2):
+            self.solver = KratosParticle.MPM2D(self.model_part1, self.model_part2, self.model_part3, self.linear_solver, self.new_element, self.move_mesh_flag, self.solver_type, self.geometry_element, self.number_particle, self.block_builder, self.pressure_dofs)
+        else:
+            self.solver = KratosParticle.MPM3D(self.model_part1, self.model_part2, self.model_part3, self.linear_solver, self.new_element, self.move_mesh_flag, self.solver_type, self.geometry_element,  self.number_particle, self.block_builder, self.pressure_dofs)
+      
+        # Set echo level
+        self._set_echo_level()
+        
+        # Check if everything is assigned correctly
+        self._check()
+      
+        if self._is_printing_rank():
+            KratosMultiphysics.Logger.PrintInfo("ParticleMPMSolver","Solver is initialized correctly.")
+
+    def Solve(self):
+        (self.solver).Solve()
+
+
+
+    ### Solver private functions
+    def _add_variables_to_model_part(self, model_part):
+        # Add displacements
+        model_part.AddNodalSolutionStepVariable(KratosMultiphysics.DISPLACEMENT)
+        
+        # Add dynamic variables
+        model_part.AddNodalSolutionStepVariable(KratosMultiphysics.VELOCITY)
+        model_part.AddNodalSolutionStepVariable(KratosMultiphysics.ACCELERATION)
+        model_part.AddNodalSolutionStepVariable(KratosMultiphysics.PRESSURE)
+        
+        # Add reactions for the displacements
+        model_part.AddNodalSolutionStepVariable(KratosMultiphysics.REACTION)
+        
+        # Add nodal force variables
+        model_part.AddNodalSolutionStepVariable(KratosMultiphysics.INTERNAL_FORCE)
+        model_part.AddNodalSolutionStepVariable(KratosMultiphysics.EXTERNAL_FORCE)
+        #model_part.AddNodalSolutionStepVariable(KratosMultiphysics.CONTACT_FORCE)
+        
+        # Add specific variables for the problem conditions
+        model_part.AddNodalSolutionStepVariable(KratosMultiphysics.POSITIVE_FACE_PRESSURE)
+        model_part.AddNodalSolutionStepVariable(KratosMultiphysics.PRESSURE)
+        model_part.AddNodalSolutionStepVariable(KratosMultiphysics.VOLUME_ACCELERATION)
+        model_part.AddNodalSolutionStepVariable(KratosMultiphysics.NODAL_MASS)
+        model_part.AddNodalSolutionStepVariable(KratosParticle.NODAL_MOMENTUM)
+        model_part.AddNodalSolutionStepVariable(KratosParticle.NODAL_INERTIA)
+        model_part.AddNodalSolutionStepVariable(KratosParticle.DISPLACEMENT_AUX)
+        model_part.AddNodalSolutionStepVariable(KratosParticle.AUX_VELOCITY)
```

Comment:
Do you need all this variables to be historical?, that 's a lot of variables and memory

Reply:
Yes @loumalouomega, I need those

Reply:
Hmmm, I will investigate this further then, as I said, I will optimize the new solver further and make a separate PR.